keel-api/src/test/kotlin/com/netflix/spinnaker/keel/rest/DeliveryConfigControllerTests.kt

```diff
@@ -257,8 +266,63 @@ internal class DeliveryConfigControllerTests : JUnit5Minutests {
                 ?.constraints
             ).isNotNull()
           }
+
+          context("no environments with stateful constraints have been evaluated") {
+            test("no constraint state") {
+              getProdConstraintStates(contentType)
+                .andExpect(status().isOk)
+                .andExpect(content().string(containsString("[]")))
```

Comment:
Should we parsing the content into a map and checking the specific fields instead of comparing strings? I'm genuinely asking to understand what pattern we're using (if any).

Reply:
Updated to selectively deserialize depending on the contentType. In this case, I wanted an assertion that would apply for yaml and json response bodies without being concerned for which was which but if you're looking for patterns to follow, that wasn't a good one.